pkg/client/imagestreams.go

```diff
@@ -109,7 +110,7 @@ func (c *imageStreams) UpdateStatus(stream *imageapi.ImageStream) (result *image
 // will be returned if no actual import was requested (the to fields were not set), or an ImageStream if import was requested.
 func (c *imageStreams) Import(isi *imageapi.ImageStreamImport) (*imageapi.ImageStreamImport, error) {
 	result := &imageapi.ImageStreamImport{}
-	if err := c.r.Post().Namespace(c.ns).Resource("imageStreamImports").Body(isi).Do().Into(result); err != nil {
+	if err := c.r.Post().Namespace(c.ns).Resource("imageStreamImports").Timeout(2 * time.Minute).Body(isi).Do().Into(result); err != nil {
```

Comment:
Yes, request timeout is set and verified inside `WithTimeoutForNonLongRunningRequests` filter, but [resthandler.go](https://github.com/openshift/origin/blob/612dcfb20d42755959895cc19285b429864ddeba/vendor/k8s.io/kubernetes/pkg/apiserver/resthandler.go#L350) has additional timeout which needs to be set here explicitly. Personally, I'd prefer the value from `request-timeout` flag would be applied to this, as well.